### PR TITLE
feat(gradle): support @<ext> file extentions

### PR DESCRIPTION
**Pull request by t-kulmburg**
## Changes:

Before this change, gradle extentions, where the file type is defined behind the version, were not recognized. This PR adds this functionality, but the file extention is not really used to search the dependency.
Dependencies with such a qualifier look like this: `foo.bar:baz:1.2.3@ext`

## Context:

[Issue 11324](https://github.com/renovatebot/renovate/issues/11324)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
Please sign CLA, see CLA comment

**Comment by t-kulmburg** [comment]
I just signed the CLA

File: lib/manager/gradle/shallow/extract.spec.ts [hunk 1]
```diff
@@ -67,7 +67,79 @@ describe('manager/gradle/shallow/extract', () => {
       },
     ]);
   });
+  it('works with file-ext', async () => {
```
Comment (viceice) [inline lib/manager/gradle/shallow/extract.spec.ts:70]:
```suggestion

  it('works with file-ext', async () => {
```

Readabillity

File: lib/manager/gradle/shallow/extract.spec.ts [hunk 2]
```diff
@@ -67,7 +67,79 @@ describe('manager/gradle/shallow/extract', () => {
...
+        ],
+      },
+      {
+        datasource: 'maven',
+        deps: [],
+        packageFile: 'settings.gradle',
+      },
+    ]);
+  });
+  it('works with file-ext-var', async () => {
```
Comment (viceice) [inline lib/manager/gradle/shallow/extract.spec.ts:108]:
```suggestion

  it('works with file-ext-var', async () => {
```

File: lib/manager/gradle/shallow/extract.spec.ts [hunk 3]
```diff
@@ -67,7 +67,79 @@ describe('manager/gradle/shallow/extract', () => {
...
+      },
+      { packageFile: 'build.gradle', deps: [] },
+      {
+        datasource: 'maven',
+        deps: [],
+        packageFile: 'settings.gradle',
+      },
+    ]);
+  });
   it('inherits gradle variables', async () => {
```
Comment (viceice) [inline lib/manager/gradle/shallow/extract.spec.ts:143]:
```suggestion

  it('inherits gradle variables', async () => {
```

**Review by zharinov — APPROVED** [review]
Looks good to me

**Review by viceice — APPROVED** [review]

**Pull request merged**